AsciiPixel/src/colorant2ansi.jl

```diff
@@ -28,16 +28,289 @@ function (enc::TermColor256)(c::AbstractRGB)
     r, g, b = clamp01nan(red(c)), clamp01nan(green(c)), clamp01nan(blue(c))
     r24, g24, b24 = map(c->round(Int, c * 23), (r, g, b))
     if r24 == g24 == b24
-        # Use grayscales because of higher resolution
-        # This way even grayscale RGB images look good.
+        # RGB scale color code
+        r24 == 0 && return 17   # 0x000000
+        r24 == 9 && return 60   # 0x5f5f5f
+        r24 == 12 && return 103 # 0x878787
+        r24 == 16 && return 146 # 0xafafaf
+        r24 == 19 && return 189 # 0xd7d7d7
+        r24 == 23 && return 232 # 0xffffff
+        # gray scale color code
         232 + r24
     else
-        r6, g6, b6 = map(c->round(Int, c * 5), (r, g, b))
-        16 + 36 * r6 + 6 * g6 + b6
+        r6, g6, b6 = map(c->floor(Int, c * 5), (r, g, b))
+        17 + 36 * r6 + 6 * g6 + b6
     end
 end
-(enc::TermColor256)(c::AbstractGray) = round(Int, 232 + clamp01nan(real(c)) * 23)
+function (enc::TermColor256)(c::AbstractGray)
+    val = round(Int, clamp01nan(gray(c)) * 26)
+    val == 0 && return 17   # 0x000000
+    val > 24 && return 232  # 0xffffff
+    return 232 + val
```

Comment:
This can be implemented wrong here; just that this version passes the `enc(enc(c)) = c` test for those 256 colors.